### PR TITLE
✨PROS 4: Make Rotation Settings Object Oriented

### PR DESCRIPTION
**Pull request by aaryan-gautam**
#### Summary:

This allows rotation, serial and vision device settings to have multiple configurations. Device items are stored inside of classes.
#### Motivation:

It is moving pros towards a more object oriented direction and allows users to have different device configurations for different device objects.
##### References (optional):

#372
#### Test Plan:

- [ ] test item
Create two rotation, vision and serial objects with different configurations and push them to the same device.

### CODE REVIEW

**Comment by WillXuCodes** [comment]
Write a more in depth test plan please

**Comment by aaryan-gautam** [comment]
> Write a more in depth test plan please

Added test plan.

**Comment by noam987** [comment]
Rotation Sensor Passed hardware testing. 
IMU did not seem to pass my initial tests, as I couldn't get any reading to display. 
Did not know how to test serial or vision

File: include/pros/imu.hpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/**
+    /**
```
Comment (WillXuCodes) [inline include/pros/imu.hpp:1]: remove this tab plase

File: include/pros/rotation.hpp [hunk 2]
```diff
@@ -25,6 +25,7 @@
 #include <iostream>
 
 #include "pros/rotation.h"
+#include "rtos.hpp"
```
Comment (WillXuCodes) [inline include/pros/rotation.hpp:28]: is this necessary? if not we should prob remove this.

**Review by WillXuCodes — CHANGES_REQUESTED** [review]

File: include/pros/rotation.hpp (hunk 2 above)
Comment (ColexDev) [reply include/pros/rotation.hpp:28]: Yes, this is used for pros::Mutex

Comment (ColexDev) [reply include/pros/rotation.hpp:28]: Yes, this is used for pros::Mutex.

**Comment by WillXuCodes** [comment]
Also make both your PRs more specific what devices and what exactly you're doing.

**Review by Richard-Stump — APPROVED** [review]

**Comment by Richard-Stump** [comment]
Fix the merge conflicts, and it should be good to go

**Comment by noam987** [comment]
Has this been hardware tested this semester? If not can we retest before we merge it.

**Comment by WillXuCodes** [comment]
> Has this been hardware tested this semester? If not can we retest before we merge it.

needs HW testing

**Comment by noam987** [comment]
This Doesn't work how we thought it would. Thus we are closing this PR

**Pull request closed without merging**